### PR TITLE
extensive changes to support DALI-1.1 datatypes

### PR DESCRIPTION
**Pull request by pdowler**

### CODE REVIEW

File: cadc-dali/src/main/java/ca/nrc/cadc/dali/tables/votable/VOTableUtil.java [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+public abstract class VOTableUtil 
+{
+    private static final Logger log = Logger.getLogger(VOTableUtil.class);
+
+    private VOTableUtil() { }
+    
+    public static int[] getArrayShape(String arraysize)
+    {
+        if (arraysize == null || arraysize.equals("1")) // interpretation TBD
+            return null;
```
Comment (andamian) [inline cadc-dali/src/main/java/ca/nrc/cadc/dali/tables/votable/VOTableUtil.java]: Don't you want to throw an exception here? Or null array shape is a valid case?

File: cadc-dali/src/main/java/ca/nrc/cadc/dali/util/DoubleArray2DFormat.java [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+        
+        StringBuilder sb = new StringBuilder();
+        for (int i=0; i<object.length; i++)
+            for (int j=0; j<object[i].length; j++)
+            {
+                sb.append(Double.toString(object[i][j]));
+                sb.append(" ");
+            }
+        return sb.substring(0, sb.length() - 1); // trim trailing comma
+    }
```
Comment (andamian) [inline cadc-dali/src/main/java/ca/nrc/cadc/dali/util/DoubleArray2DFormat.java]: trailing space hence a trim() would be more suitable maybe.

File: cadc-dali/src/main/java/ca/nrc/cadc/dali/tables/votable/VOTableUtil.java (hunk 1 above)
Comment (pdowler) [reply cadc-dali/src/main/java/ca/nrc/cadc/dali/tables/votable/VOTableUtil.java]: null arrayShape means it is a scalar value

File: cadc-dali/src/main/java/ca/nrc/cadc/dali/util/DoubleArray2DFormat.java (hunk 2 above)
Comment (pdowler) [reply cadc-dali/src/main/java/ca/nrc/cadc/dali/util/DoubleArray2DFormat.java]: I changed a bunch of array formatters to use StringBuilder substring to trimToSize and toString; the separator used to be comma hence the old trim hack :-)

**Pull request merged**